server/src/test/java/org/elasticsearch/search/suggest/SuggestSearchIT.java

```diff
@@ -1005,6 +1005,45 @@ public void testSuggestWithFieldAlias() throws Exception {
         assertSuggestion(searchSuggest, 0, "suggestion", "apple");
     }
 
+    public void testMinDocFreq() throws Exception {
```

Comment:
This test seems to pass even without the above changes in DirectCandidateGenerator. Since it uses the TermSuggester is looksl like it doesn't trigger the #drawCandidates codepath so I think it need to be changed a bit.

Reply:
Good catch, the terms suggester already handles min doc freq so I changed the test to check the phrase suggester instead.